src/main/java/conflux/dex/service/blockchain/BlockchainSettlementService.java

```diff
@@ -201,13 +211,51 @@ private void settle(Settleable data) throws Exception {
 			// 2. service restarted and last item not settled on chain yet
 			this.validateNonce(data, false);
 			this.sendTransaction(data, context);
+			txHash = data.getSettledTxHash();
 		}
 		
 		data.updateSettlement(this.dao, SettlementStatus.OnChainSettled);
 		
 		this.monitor.add(data);
+
+		if (txHash != null) {
+			waitTxExecuted(data);
+		}
 	}
-	
+
+	private void waitTxExecuted(Settleable data) {
+		String txHash = data.getSettledTxHash();
+		while (true) {
+			// maybe it has been paused by tx monitor already.
+			if (this.healthService.getPauseSource().isPresent()) {
```

Comment:
What if pause source is "Manual"?

Reply:
Changed to `if (this.healthService.isPausedBy(PauseSource.Blockchain))`